packages/patterns/src/components/create-pattern-modal.js

```diff
@@ -74,13 +74,17 @@ export function CreatePatternModalContents( {
 
 	const { categoryMap, findOrCreateTerm } = useAddPatternCategory();
 
+	const validation = useValidation();
+
 	const fields = [
 		{
 			id: 'title',
 			label: __( 'Name' ),
 			type: 'text',
-			isValid: ( value ) => {
-				return value.title.length > 0;
+			validationSchema: {
+				minLength: 1,
+				maxLength: 10,
+				onTouched: true,
```

Comment:
In relation to @youknowriad [comment above](https://github.com/WordPress/gutenberg/pull/67357/files#r1870895866), it feels a bit weird to me to add these constraints as part of the field definition. Especially if we may use some of these same definitions across post types. For example would `title` validation be the same across a pattern, template, or category?

I did prefer to have something like `validationSupports` for a specific field definition or field type ( if needed, it may be irrelevant unless we make use of some of the native validation ).

I would lean towards something similar to the JSON schema, where we link the validation conditions to the `key`. In this case we may pass something like this to the `useValidation` hook:
```
{
  "title": {
            "minLength": 1,
            "maxLength": 10,
            "onTouched": true
        }
}
```
The only questionable one being `onTouched`, that may be something better suited for the field definition.

Reply:
Foe me maxLength and minLength are not just validation constraints, they are tools to understand the field itself, the UI to edit it... They should be part of the field definition.

Reply:
> Foe me maxLength and minLength are not just validation constraints, they are tools to understand the field itself, the UI to edit it... They should be part of the field definition.

@youknowriad should we possibly separate attributes (including maxLength minLength) from validation schema? If something is needed on the field UI and such, it should be on the attributes level. This means supporting attributes for fields + schema.